src/App.tsx

```diff
@@ -15,36 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphaberticaly = 'Sort alphabetically',
+  Length = 'Sort by length',
+  Reverse = 'Reverse',
+}
+
+const SORT_FIELD_ALPHABET: SortType = SortType.Alphaberticaly;
+const SORT_FIELD_LENGTH: SortType = SortType.Length;
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField }: { sortField: SortType },
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SORT_FIELD_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+
+  let visibleGoods = sortField
+    ? getPreparedGoods(goodsFromServer, { sortField })
+    : [...goodsFromServer];
+
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
```

Comment:
[CODE KNOWLEDGE] - Directly mutating the state like reversing the `visibleGoods` array is not recommended in React. Instead, you should use the `useState` hook to set the reversed state of the goods.